.circleci/wait-until-previous-builds-complete.sh

```diff
@@ -0,0 +1,165 @@
+#! /bin/bash
+
+set -e
+set -o pipefail
+set -o functrace
+
+#********************
+# VARIABLES
+# *******************
+api_root="https://circleci.com/api/v2/"
+# polling branch name. Other branches (PRs) are excluded from consideration.
+branch="master"
+# Polling workflow name. Other workflow names (nightly-tests and deploy-staging) are excluded from consideration.
+workflow_name="build-test-deploy"
+project_slug="gh/all-of-us"
+
+# Max check time on a workflow is 45 minutes because e2e tests may take a long time to finish.
+# DISCLAIMER This max time may not be enough if there are 2 or more workflows in waiting before this workflow.
+max_time=$((45 * 60))
+sleep_time=30
+
+#********************
+# FUNCTIONS
+# *******************
+
+# CIRCLECI_TOKEN is a env variable whose value is personal token.
```

Comment:
Am I understanding correctly that for this approach to work, you needed to create a personal API token and upload it into Circle? This basically turns your personal token into a shared credential which is really not good - we should look for other solutions if I've understood this correctly.

Reply:
This is too insecure, I don't think we can move forward with this approach. Could you look at other options for authenticating these calls? I can investigate a bit later today as well.